### PR TITLE
Multiple code quality improvements

### PR DESCRIPTION
**Pull request by Secrus**
- Added a few `flake8` plugins and corrected issues that raised
- Added `pyupgrade` (set to 3.6+)
- Added `pycln` linter

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -19,3 +19,40 @@ repos:
...
+    hooks:
+      - id: pyupgrade
+        args: [--py36-plus]
+
+  - repo: https://github.com/hadialqattan/pycln
+    rev: v2.0.1
+    hooks:
+      - id: pycln
+        args: [--all]
+        exclude: ^.*__init__\.py$ # without this, pycln removes TOMLDocument import
```
Comment (frostming) [inline .pre-commit-config.yaml:52]: Add it to `__all__` should fix it

File: tomlkit/exceptions.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
+from __future__ import annotations
```
Comment (frostming) [inline tomlkit/exceptions.py:1]: This only works on 3.7+ IIRC?

**Review by frostming — CHANGES_REQUESTED** [review]

File: tomlkit/exceptions.py (hunk 2 above)
Comment (Secrus) [reply tomlkit/exceptions.py:1]: Oh, right. I knew I forgot about something. I guess I will need to do some reverting.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Secrus) [reply .pre-commit-config.yaml]: Fixed

File: tomlkit/exceptions.py (hunk 2 above)
Comment (Secrus) [reply tomlkit/exceptions.py]: Reverted.

**Review by frostming — APPROVED** [review]
LGTM

**Pull request merged**